src/librbd/crypto/luks/FormatLUKSRequest.h

```diff
@@ -0,0 +1,51 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_LIBRBD_CRYPTO_LUKS_FORMAT_LUKS_REQUEST_H
+#define CEPH_LIBRBD_CRYPTO_LUKS_FORMAT_LUKS_REQUEST_H
+
+#include <libcryptsetup.h>
+#include "librbd/ImageCtx.h"
+#include "librbd/crypto/CryptoInterface.h"
+
+namespace librbd {
+
+class ImageCtx;
+
+namespace crypto {
+namespace luks {
+
+template <typename I>
+class FormatLUKSRequest : public Context {
```

Comment:
We try to avoid basing non-trivial state machines on `Context` and instead handle the `delete this` directly in a `finish(int)` method:
```suggestion
class FormatLUKSRequest {
```